gen/parser.go

```diff
@@ -0,0 +1,104 @@
+package gen
+
+import (
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"go/types"
+
+	"github.com/TwiN/go-color"
+)
+
+var (
+	functions = make(map[string]Function)
+	resources = make(map[string]struct{}) // this is a set.
+)
+
+func GetFunctions() map[string]Function {
+	return functions
+}
+
+func GetResources() map[string]struct{} {
+	return resources
+}
+
+func parse() error {
+	packages, err := parser.ParseDir(token.NewFileSet(), ".", nil, parser.ParseComments)
+	if err != nil {
+		return err
+	}
+	for _, file := range packages["opslevel"].Files {
+		ast.FileExports(file)
+		ast.Inspect(file, func(n ast.Node) bool {
+			var (
+				funcName    string
```

Comment:
Can these `var` declarations be moved up out of the for loop? They are only be declared once either way and it's less to read through within the `for` loop.